extractor/src/main/java/com/example/extractor/ConfigSerializer.java

```diff
@@ -302,6 +302,38 @@ public static String serializeAllInOneConfigToString(
         return TemplateProcessor.replacePlaceholders(template, combinedModuleContent, isTreeWalker);
     }
 
+    /**
+     * Creates a deep copy of the given configuration.
+     *
+     * @param config the configuration to copy
+     * @param newId the new ID to assign to the copied configuration
+     * @return a new {@link Configuration} that is a deep copy of the provided configuration
+     * @throws CheckstyleException if copying fails
+     */
+    private static Configuration copyConfiguration(final Configuration config, final String newId) {
+        final DefaultConfiguration newConfig = new DefaultConfiguration(config.getName());
+
+        for (final String name : config.getPropertyNames()) {
+            try {
+                final String value = config.getProperty(name);
+                newConfig.addProperty(name, value);
+            }
+            catch (CheckstyleException ex) {
+                // Property not found, skipping
+            }
+        }
+
+        // Set the new ID
+        newConfig.addProperty("id", newId);
```

Comment:
> Put this to loop over properties.

```
    // Set the new ID once, after copying existing properties
    newConfig.addProperty("id", newId);
```

This implementation is correct because:

1. It copies all existing properties from the original configuration.
2. It adds the new ID property once, after copying existing properties. This ensures that each module gets exactly one ID property.


If i add it to the loop, i get something like this:
![image](https://github.com/user-attachments/assets/ba6ff616-15e6-45cd-b879-9b99883e8363)

```
    private static Configuration copyConfiguration(final Configuration config, final String newId) {
        final DefaultConfiguration newConfig = new DefaultConfiguration(config.getName());

        for (final String name : config.getPropertyNames()) {
            try {
                final String value = config.getProperty(name);
                newConfig.addProperty(name, value);
            }
            catch (CheckstyleException ex) {
                // Property not found, skipping
            }
        }

        // Set the new ID
        newConfig.addProperty("id", newId);

        // Copy child configurations
        for (final Configuration child : config.getChildren()) {
            newConfig.addChild(copyConfiguration(child, newId + "-child"));
        }

        return newConfig;
    }
```